sdk/webpubsub/Azure.Messaging.WebPubSub.Client/src/WebPubSubClient.cs

```diff
@@ -56,7 +112,27 @@ public WebPubSubClient(Uri clientAccessUri, WebPubSubClientOptions options) : th
         /// <param name="options">A option for the client.</param>
         public WebPubSubClient(WebPubSubClientCredential credential, WebPubSubClientOptions options = null)
         {
-            throw new NotImplementedException();
+            _webPubSubClientCredential = credential ?? throw new ArgumentNullException(nameof(credential));
+
+            if (options == null)
+            {
+                _options = new WebPubSubClientOptions();
+            }
+            else
+            {
+                _options = options;
+            }
+            _protocol = _options.Protocol ?? throw new ArgumentNullException(nameof(options));
+            WebSocketClientFactory = new WebSocketClientFactory();
+
+            _clientState = new ClientState();
+
+            _reconnectRetryPolicy = new WebPubSubRetryPolicy(_options.ReconnectRetryOptions);
```

Comment:
`_options.ReconnectRetryOptions` : could it be null?

Reply:
Can't be null, it's set in ctor in WebPubSubClientOption and can be reassign